neo4j/src/test/java/org/hibernate/ogm/datastore/neo4j/utils/Neo4jTestHelper.java

```diff
@@ -137,6 +151,50 @@ private static DatastoreProvider getDatastoreProvider(SessionFactory sessionFact
 		throw new RuntimeException( "Not testing with Neo4jDB, cannot extract underlying provider" );
 	}
 
+	public static List<Neo4jIndexSpec> getIndexes(DatastoreProvider provider) {
+		if ( EmbeddedNeo4jDatastoreProvider.class.isInstance( provider ) ) {
+			return getEmbeddedNeo4jIndexes( provider );
+		}
+		if ( HttpNeo4jDatastoreProvider.class.isInstance( provider ) ) {
+			return getHttpNeo4jIndexes( provider );
+		}
+		if ( BoltNeo4jDatastoreProvider.class.isInstance( provider ) ) {
+			return getBoltIndexes( provider );
+		}
+		return Collections.emptyList();
+	}
+
+	private static List<Neo4jIndexSpec> getEmbeddedNeo4jIndexes(DatastoreProvider provider) {
+		GraphDatabaseService neo4jDb = EmbeddedNeo4jDatastoreProvider.class.cast( provider ).getDatabase();
+		try ( Transaction tx = neo4jDb.beginTx() ) {
+			List<Neo4jIndexSpec> indexes = toStream( neo4jDb.schema().getIndexes() )
+					.map( i -> new Neo4jIndexSpec( i.getLabel(), toList( i.getPropertyKeys() ), i.isConstraintIndex() ) )
+					.collect( Collectors.toList() );
+			tx.success();
+			return indexes;
+		}
+	}
+
+	private static List<Neo4jIndexSpec> getHttpNeo4jIndexes(DatastoreProvider provider) {
+		// todo: neo4jHttpClient does not return graph values
```

Comment:
@DavideD , I'm trying to get indexes by CALL db.indexes via HttpNeo4jClient. However it returns
```
{
  "results" : [ {
    "columns" : [ "description", "label", "properties", "state", "type", "provider" ],
    "data" : [ {
      "row" : null,
      "graph" : {
        "nodes" : [ ],
        "relationships" : [ ]
      },
      "meta" : null
    },... ],
  "errors" : [ ]
}
```

in contrast of direct rest response

```
{
    "results": [
        {
            "columns": [ "description",  "label", "properties", "state", "type", "provider" ],
            "data": [
                {
                    "row": [
                        "INDEX ON :Neo4jIndexTest$Person(age)",
                        "Neo4jIndexTest$Person",
                        [
                            "age"
                        ],
                        "ONLINE",
                        "node_label_property",
                        {
                            "version": "1.0",
                            "key": "lucene+native"
                        }
                    ]
                }, ...
    ],
    "errors": []
}
```

do we have any filter somewhere?

Reply:
Never mind, I have found my mistake. There are two modes for the query to work. I changed to `row` mode and everything works as expected.